storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -1103,21 +1216,39 @@
      (.set_reset_log_level_timeout_epoch log-config (coerce/to-long timeout))
      (.unset_reset_log_level_timeout_epoch log-config))))
 
+(defmethod blob-sync-code :distributed [conf nimbus]
+  (if (not (is-leader nimbus :throw-exception false))
+    (let [storm-cluster-state (:storm-cluster-state nimbus)
+          nimbus-host-port-info (:nimbus-host-port-info nimbus)
+          blob-store-key-list (get-key-list-from-blob-store (:blob-store nimbus))
+          zk-key-list (into [] (.blobstore storm-cluster-state (fn [] (blob-sync-code conf nimbus))))]
+      (log-debug "blob-sync-code " "keys " blob-store-key-list "blob-key " blob-store-key-list)
+      (let [sync-object (doto
+                          (SyncBlobs. (:blob-store nimbus) conf)
+                          (.setNimbusInfo nimbus-host-port-info)
+                          (.setBlobStoreKeyList (if (not-nil? blob-store-key-list) (java.util.ArrayList. blob-store-key-list) (java.util.ArrayList.)))
+                          (.setZookeeperKeyList (if (not-nil? zk-key-list) (java.util.ArrayList. zk-key-list) (java.util.ArrayList.))))]
```

Comment:
Minor, but I think we should import java util ArrayList instead of repeating complete qualified name